cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -0,0 +1,342 @@
+package dpdk
+
+import (
+	"context"
+	"fmt"
+	sriovv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	sriovcluster "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	sriovnamespaces "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/networks"
+	utilNodes "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/performanceprofile"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/pods"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
+
+	var numa0DeviceList []*sriovv1.InterfaceExt
+	var numa1DeviceList []*sriovv1.InterfaceExt
+
+	BeforeAll(func() {
+		if discovery.Enabled() {
+			Skip("Discovery mode not supported")
+		}
+
+		isSNO, err := utilNodes.IsSingleNodeCluster()
+		Expect(err).ToNot(HaveOccurred())
+		if isSNO {
+			Skip("Single Node openshift not yet supported")
+		}
+
+		err = namespaces.Create(sriovnamespaces.Test, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Clean SRIOV policies and networks")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovCapableNodes, err := sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(len(sriovCapableNodes.Nodes)).To(BeNumerically(">", 0))
+		testNode, err := nodes.GetByName(sriovCapableNodes.Nodes[0])
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Apply single-numa-node performance profile")
+		perfProfile := performanceprofile.DefineSingleNUMANode("single-numa-node-pp", machineConfigPoolName)
+		err = performanceprofile.OverridePerformanceProfile("single-numa-node-pp", machineConfigPoolName, perfProfile)
+		Expect(err).ToNot(HaveOccurred())
+
+		sriovDevices, err := sriovCapableNodes.FindSriovDevices(testNode.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		numa0DeviceList, err = findDevicesOnNUMANode(testNode, sriovDevices, "0")
+		Expect(err).ToNot(HaveOccurred())
+		numa1DeviceList, err = findDevicesOnNUMANode(testNode, sriovDevices, "1")
```

Comment:
we need to skip the tests if the node has only one numa not failed and that can be the case for RAN servers